src/lib/storage/table.cpp

```diff
@@ -87,5 +88,31 @@ void Table::_append_new_chunk() {
   _chunks.push_back(std::move(new_chunk));
 }
 
-void Table::compress_chunk(ChunkID chunk_id) { throw std::runtime_error("Implement Table::compress_chunk"); }
+void Table::compress_chunk(ChunkID chunk_id) {
+  const auto& uncompressed_chunk = get_chunk(chunk_id);
+  Chunk compressed_chunk = Chunk();
+
+  std::vector<std::future<std::shared_ptr<BaseSegment>>> compressed_segment_futures;
+
+  const auto col_count = column_count();
+  for (ColumnID column_id = ColumnID{0}; column_id < col_count; ++column_id) {
+    const auto uncompressed_segment = uncompressed_chunk.get_segment(column_id);
+    std::promise<std::shared_ptr<BaseSegment>> promise;
+    compressed_segment_futures.push_back(promise.get_future());
+    std::thread thread(_compress_segment, std::move(promise), column_type(column_id), uncompressed_segment);
+    thread.detach();
```

Comment:
The [std::thread destructor](https://en.cppreference.com/w/cpp/thread/thread/~thread) will call terminate() on the thread when it's dtor is called unless we do this. Since the thread is about to end on it's own anyway, we thought doing the shutdown cooperatively might be nicer (and may help us find bugs quicker?) and adding one line does not decrease readability.

Would you rather not have that line?